lib/payment_processor/clients/braintree/transaction.rb

```diff
@@ -3,18 +3,33 @@ module Clients
     module Braintree
       class Transaction
 
-        def self.make_transaction(nonce:, amount:, user:)
-          new(nonce, amount, user).sale
+        def self.make_transaction(nonce:, amount:, user:, store: nil)
+          new(nonce, amount, user, store).sale
```

Comment:
Store is a class or module constant. By optionally injecting the store into `Transaction`, we decouple `Transaction` from the whole mess of storage, nor must it commit to a specific type of persistence. If no store constant is passed, then nothing gets persisted locally.